src/objectExplorer/nodes/connectionNode.ts

```diff
@@ -59,6 +62,48 @@ export class ConnectionNode extends TreeNodeInfo {
         } else {
             this.iconPath = ObjectExplorerUtils.iconPath(ICON_SERVER_DISCONNECTED);
         }
+
+        // Tooltip logic: show all non-default properties except those in the label (database, user, server)
+        this.tooltip = this.getConnectionTooltip(connectionProfile);
+    }
+
+    private getConnectionTooltip(connectionProfile: IConnectionProfile): string {
+        // Properties to exclude
+        const exclude = ["database", "user", "server", "profileName", "id", "groupId"];
+        // Default values for comparison
+        const defaultValues = {
+            encrypt: "Mandatory",
+            trustServerCertificate: false,
+            persistSecurityInfo: false,
+            azureAuthType: AzureAuthType.AuthCodeGrant,
+            multipleActiveResultSets: false,
+            connectTimeout: Constants.defaultConnectionTimeout,
+            commandTimeout: Constants.defaultCommandTimeout,
+            applicationName: Constants.connectionApplicationName,
+            savePassword: false,
+            emptyPasswordInput: false,
+            profileSource: 0,
+        };
+
+        let props: string[] = [];
+
+        Object.keys(connectionProfile).forEach((key) => {
+            const value = (connectionProfile as any)[key];
+            if (exclude.includes(key)) {
+                return;
+            }
+            if (!value || value === "") {
+                return;
+            }
+
+            if (key in defaultValues && value === defaultValues[key]) {
+                return;
+            }
+
+            props.push(`${key}: ${value}`);
+        });
+
+        return props.length > 0 ? props.join("\n") : "";
```

Comment:
if props is empty, will this show an empty tooltip, or the default tooltip (just the display text of the node)?